packages/terra-date-picker/src/react-datepicker/calendar.jsx

```diff
@@ -397,16 +403,16 @@ export default class Calendar extends React.Component {
 
   changeYear = (year) => {
     this.setState({
-      date: setYear(cloneDate(this.state.date), year)
+      date: setYear(cloneDate(this.state.date), year).startOf('month')
     })
-    this.props.setPreSelection(setYear(cloneDate(this.state.date), year));
+    this.props.setPreSelection(setYear(cloneDate(this.state.date), year).startOf('month'));
```

Comment:
Looks like the preselected date is not getting the dotted line focus style.

Reply:
I was using the mouse click to navigate to the previous/next month and after consulting @neilpfeiffer the preselected date of the first of the month should _not_ have the focus style when navigating using the mouse. When I use the keyboard to interact with the picker, then the preselected date gets the focus. So this part is working as expected.

This scenario doesn't appear to be working as expected.
When I launched the picker by clicking on the calendar button, the preselected date is not focused (which is expected). However, when I start using the arrow keys to navigate within the calendar, the preselected date should display but it is not showing. If I hit the right arrow twice, it would jump to the next month but then hitting the left or right arrow again doesn't do anything.

Reply:
Fixed 251400f